### PR TITLE
Definitions can now have nested paths

### PR DESCRIPTION
**Pull request by wtrocki**
Closes # 

### Verification Steps

1. Do x
2. Do y

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [ ] Other (please specify)

### CODE REVIEW

File: pkg/surveyjson/survey.go [hunk 1]
```diff
@@ -352,8 +352,29 @@ func (o *JSONSchemaOptions) recurse(name string, prefixes []string, requiredFiel
...
+			}
+
+			currentObject = object.(map[string]interface{})
+		}
+
+		var mainDefinition *JSONSchemaType
+
+		nestedJSON, err := json.Marshal(currentObject)
+		if err != nil {
+			return err
```
Comment (wtrocki) [inline pkg/surveyjson/survey.go]: Btw.. let's wrap error like I have been doing above. Returning just error will return horrible message or no message at all.

Comment (jackdelahunt) [reply pkg/surveyjson/survey.go]: will do that and merge

File: pkg/surveyjson/survey.go [hunk 2]
```diff
@@ -352,8 +352,29 @@ func (o *JSONSchemaOptions) recurse(name string, prefixes []string, requiredFiel
...
-		// TODO ref path would need to be traversed to get sub objects, assuming right now that all defs are one level deep
-		err = o.recurse(name, make([]string, 0), make([]string, 0), definitions[refPath[0]], nil, output, make([]survey.Validator, 0), existingValues, definitions)
+		currentObject := (*definitions[refPath[0]]).(map[string]interface{})
+		for i := 1; i < len(refPath); i += 1 {
+			object, ok := currentObject[refPath[i]]
+			if !ok {
+				return errors.New(fmt.Sprintf("Could not resolve ref path \"%v\" is not a key in the object", refPath[i]))
+			}
+
+			currentObject = object.(map[string]interface{})
```
Comment (wtrocki) [inline pkg/surveyjson/survey.go:362]: Note that we should have extra variable `ok` to check if cast even works. 
Without that we get nil pointer reference.

File: pkg/surveyjson/survey.go [hunk 3]
```diff
@@ -352,8 +352,29 @@ func (o *JSONSchemaOptions) recurse(name string, prefixes []string, requiredFiel
...
+		}
+
+		var mainDefinition *JSONSchemaType
+
+		nestedJSON, err := json.Marshal(currentObject)
+		if err != nil {
+			return errors.New(fmt.Sprintf("Cannot marshal to json object %v", currentObject))
```
Comment (wtrocki) [inline pkg/surveyjson/survey.go:369]: errors.wrap (we should include actual error)

File: problem.json [hunk 4]
```diff
@@ -0,0 +1,105 @@
+{
```
Comment (wtrocki) [inline problem.json:1]: This file should not be commited :)

**Pull request merged**